task6/posts.js

```diff
@@ -0,0 +1,674 @@
+        {
+            id: '17',
+            description: 'Первая строка содержит целое число n (2 ≤ n ≤ 300). Каждая из последующих n строк содержит два целых числа ai и bi (0 ≤ ai < bi ≤ 1 000 000 000) — границы интервала для силы руки каждого игрока.',
+            createdAt: new Date('2020-01-23T20:07:00'),
+            author: 'User_1',
+            likes: [],
+            hashTags: [],
+        },
+        {
+            id: '18',
+            description: 'Выведите n строк, i-я из которых содержит одно действительное число — вероятность победы в туре i-го игрока. Абсолютная погрешность не должна превосходить 10−9.',
+            createdAt: new Date('2020-01-23T20:10:27'),
+            author: 'Task',
+            photoLink: 'https://sun1.beltelecom-by-minsk.userapi.com/Uv7oJSk0ePo7QMQQZgWN1al2w0hF9FqcL5d11Q/MSdYzJobMBk.jpg',
+            likes: [],
+            hashTags: [],
+        },
+        {
+            id: '19',
+            description: 'Sample text',
+            createdAt: new Date('2020-03-23T20:09:00'),
+            author: 'User_1',
+            likes: [],
+            hashTags: ['tag_1', 'tag_2', 'tag_3', 'tag_4', 'tag_5'],
+        },
+        {
+            id: '20',
+            description: '',
+            createdAt: new Date('2020-03-23T21:10:59'),
+            author: 'Task',
+            photoLink: 'https://sun1.beltelecom-by-minsk.userapi.com/Uv7oJSk0ePo7QMQQZgWN1al2w0hF9FqcL5d11Q/MSdYzJobMBk.jpg',
+            likes: [],
+            hashTags: [],
+        },
+    ];
+
+    function getPosts(skip = 0, top = 10, filterConfig) {
+        if (typeof (skip) !== 'number') {
+            console.log('Invalid argument: "skip" isn\'t number');
+            return;
+        }
+        if (typeof top !== 'number') {
+            console.log('Invalid argument: "top" isn\'t number');
+            return;
+        }
+        if (!['object', 'undefined'].includes(typeof filterConfig)) {
+            console.log('Invalid argument: "filterConfig" isn\'t object');
+            return;
+        }
+        for (let key in filterConfig) if (!(key in posts[0]) && key !== 'createdAt') {
+            console.log(`Invalid argument: "filterConfig" contains invalid property "${key}"`);
+            return;
+        }
+
+        let res = posts.slice();
+        if (typeof filterConfig !== 'undefined' && 'createdAt' in filterConfig)
+            res = res.filter(item => (item.createdAt.getTime() <= filterConfig.createdAt.getTime() + 86400000) && item.createdAt >= filterConfig.createdAt);
+        res = res.filter(item => {
+            for (let key in filterConfig) if ((!['createdAt', 'hashTags', 'likes'].includes(key)) && filterConfig[key] !== item[key])
+                return false;
+            return true;
+        });
+        if (typeof filterConfig !== 'undefined' && 'hashTags' in filterConfig)
+            res = res.filter(item => {
+               for (let tag of filterConfig.hashTags) if (!item.hashTags.includes(tag))
```

Comment:
Can be simplified to `filterConfig.hashTags.every(tag => item.hashTags.includes(tag))`, but that's also ok